confluent_kafka/producer.py

```diff
@@ -0,0 +1,130 @@
+                 category=DeprecationWarning, stacklevel=2)
+
+        self._key_serializer = key_serializer
+        self._value_serializer = value_serializer
+
+        # Callbacks can be set in the conf dict or *ideally* as parameters.
+        # Handle both cases prior to passing along to _impl
+        # If callbacks are configured in both places parameter values take precedence.
+        if not error_cb:
+            error_cb = conf.get('error_cb', None)
+
+        if not stats_cb:
+            stats_cb = conf.get('stats_cb', None)
+
+        if not throttle_cb:
+            throttle_cb = conf.get('throttle_cb', None)
+
+        if not logger:
+            logger = conf.get('logger', None)
+
+        super(Producer, self).__init__(conf, error_cb=error_cb, stats_cb=stats_cb,
+                                       throttle_cb=throttle_cb, logger=logger)
+
+    def produce(self, topic, value=None, key=None, partition=PARTITION_UA,
+                on_delivery=None, callback=None, timestamp=0, headers=None,
+                key_serializer=None, value_serializer=None):
+        """
+            Produces message to Kafka
+
+            :param str topic: Topic to produce message to.
+            :param str|bytes|obj value: Message payload; value_serializer required for non-character values.
+            :param str|bytes|obj key: Message key payload; key_serializer required for non-character keys.
+            :param int partition: Partition to produce to, else uses the configured built-in partitioner.
+                Default value is PARTITION_UA(round-robin over all partitions).
+            :param func on_delivery(confluent_kafka.KafkaError, confluent_kafka.Message):
+                Delivery report callback to call on successful or failed delivery.
+            :param func callback(confluent_kafka.KafkaError, confluent_kafka.Message):
+                See on_delivery.
+            :param int timestamp: Message timestamp (CreateTime) in milliseconds since epoch UTC
+                (requires librdkafka >= v0.9.4, api.version.request=true, and broker >= 0.10.0.0).
+                Default value is current time.
+            :param dict|list headers: Message headers to set on the message. The header key must be a string while
+                the value must be binary, unicode or None. Accepts a list of (key,value) or a dict.
+                (Requires librdkafka >= v0.11.4 and broker version >= 0.11.0.0)
+            :param func key_serializer(topic, key): Producer key_serializer override;
+                Converts message key to bytes. **note** serializers are responsible for handling NULL keys
+            :param func value_serializer(topic, value): Producer value_serializer override;
+                Converts message value to bytes. **note** serializers are responsible for handling NULL values
+
+        """
+
+        # on_delivery is an alias for callback and take precedence
+        if callback and not on_delivery:
```

Comment:
do we have a timeline for depreciating this?

Reply:
My understanding of proper feature deprecation is one major release. i.e. Deprecation warning in 1.0 removal in 2.0.